network/p2p/gossip/gossip.go

```diff
@@ -180,6 +190,112 @@ func (p *PullGossiper[T, U]) handleResponse(
 	p.receivedBytes.Add(float64(receivedBytes))
 }
 
+// NewPushGossiper returns an instance of PushGossiper
+func NewPushGossiper[T Gossipable](client *p2p.Client, namespace string) *PushGossiper[T] {
+	return &PushGossiper[T]{
+		client: client,
+		sentN: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "gossip_sent_n",
+			Help:      "amount of gossip sent (n)",
+		}),
+		sentBytes: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "gossip_sent_bytes",
+			Help:      "amount of gossip sent (bytes)",
+		}),
+	}
+}
+
+// PushGossiper broadcasts gossip to peers randomly in the network
+type PushGossiper[T Gossipable] struct {
+	client *p2p.Client
+
+	sentN     prometheus.Counter
+	sentBytes prometheus.Counter
+
+	lock   sync.Mutex
+	queued []T
+}
+
+// Gossip flushes any queued gossipables
+func (p *PushGossiper[T]) Gossip(ctx context.Context) error {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	if len(p.queued) == 0 {
+		return nil
+	}
+
+	msg := &sdk.PushGossip{
+		Gossip: make([][]byte, 0, len(p.queued)),
+	}
+
+	sentBytes := 0
+	for _, tx := range p.queued {
+		bytes, err := tx.Marshal()
+		if err != nil {
+			return err
+		}
+
+		msg.Gossip = append(msg.Gossip, bytes)
+		sentBytes += len(bytes)
+	}
+
+	p.queued = nil
+
+	msgBytes, err := proto.Marshal(msg)
+	if err != nil {
+		return err
+	}
+
+	p.sentN.Add(float64(len(msg.Gossip)))
+	p.sentBytes.Add(float64(sentBytes))
+
+	return p.client.AppGossip(ctx, msgBytes)
+}
+
+func (p *PushGossiper[T]) Add(gossipables ...T) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	p.queued = append(p.queued, gossipables...)
+}
```

Comment:
The purpose of `Add` is to allow the user of this to pool push gossip right? (Rather than just immediately sending it)

Reply:
Yeah